demos/ret2spec_common.cc

```diff
@@ -0,0 +1,101 @@
+#include <array>
+#include <cstring>
+#include <iostream>
+#include <vector>
+
+#include "cache_sidechannel.h"
+#include "instr.h"
+#include "ret2spec_common.h"
+#include "utils.h"
+
+// We cannot include the private_data twice, because that would lead to linking
+// error. Declaring it extern in here.
+extern const char *private_data;
+
+// Calls sched_yield in the cross-address-space version.
+void (*conditionally_unschedule)();
+// Calls sched_yield in the cross-address-space version, in the
+// same-address-space version starts the ReturnsFalse recursion.
+void (*unschedule_or_start_returns_false)();
+
+// Global variables used to avoid passing parameters through recursive function
+// calls. Since we flush whole stack frames from the cache, it is important not
+// to store on stack any data that might be affected by being flushed from
+// cache.
+size_t current_offset;
```

Comment:
Same for these other global variables -- make `static` or put in an empty namespace, since they're only used in this file.

Reply:
That's incorrect. This is used cross-file. See the declarations in ret2spec_common.h